### PR TITLE
Feature/cd logging provider

### PR DESCRIPTION
**Pull request by chrisdexnimble**
**What is the change?**
To add correlation id to all log outputs, and to add correlation id to all outgoing requests to the trams-api

**Why do we need the change?**
logging across services should use a consistent correlation id to make it easier to see an end to end picture when debugging

**What is the impact?**
none

**Azure DevOps Ticket**
https://dfe-gov-uk.visualstudio.com/Academies-and-Free-Schools-SIP/_sprints/taskboard/Concerns%20Casework/Academies-and-Free-Schools-SIP/CC/CC%20-%20Iteration%2030?workitem=110751

### CODE REVIEW

File: ConcernsCaseWork/Service.TRAMS/Base/AbstractService.cs [hunk 1]
```diff
@@ -15,19 +16,21 @@ namespace Service.TRAMS.Base
...
 	public abstract class AbstractService
 	{
-		internal readonly IHttpClientFactory ClientFactory;
+		private readonly ICorrelationContext _correlationContext;
+		private readonly IHttpClientFactory _clientFactory;
 		private readonly ILogger<AbstractService> _logger;
-		internal const string HttpClientName = "TramsClient";
-		internal const string EndpointsVersion = "v2";
-		internal const string EndpointPrefix = "concerns-cases";
+		protected const string _httpClientName = "TramsClient";
```
Comment (emma-w-nimble) [inline ConcernsCaseWork/Service.TRAMS/Base/AbstractService.cs:22]: I've made some changes to this class in the decoupled version of concerns, so if it's agreed, would be putting these back to being properties:
https://github.com/DFE-Digital/amsd-casework/pull/407/files#diff-e5154a9be6c98acf69afd6dd560cad43c458a7369790bba6383192ead8fb86fa

Open to discussion, but thought it worth mentioning...

File: ConcernsCaseWork/ConcernsCaseWork/Program.cs [hunk 2]
```diff
@@ -10,12 +10,17 @@ public class Program
...
-            Log.Logger = new LoggerConfiguration()
+            var logConfig = new LoggerConfiguration()
                 .MinimumLevel.Override("Microsoft", LogEventLevel.Information)
                 .Enrich.FromLogContext()
                 .WriteTo.Console(new RenderedCompactJsonFormatter())
-				.WriteTo.Sentry()
-                .CreateLogger();
+				.WriteTo.Sentry();
+                
+	        #if DEBUG
```
Comment (mikestock-nimble) [inline ConcernsCaseWork/ConcernsCaseWork/Program.cs:19]: QUERY are these debug statements meant to be here?

**Review by mikestock-nimble — APPROVED** [review]

File: ConcernsCaseWork/ConcernsCaseWork/Program.cs (hunk 2 above)
Comment (chrisdexnimble) [reply ConcernsCaseWork/ConcernsCaseWork/Program.cs]: can't remember why so removed it

File: ConcernsCaseWork/Service.TRAMS/Base/AbstractService.cs (hunk 1 above)
Comment (chrisdexnimble) [reply ConcernsCaseWork/Service.TRAMS/Base/AbstractService.cs]: Updated them to make things easier. 
But there's no base class to set the client name, so made it 
internal string HttpClientName { get; init; } = "TramsClient"; // was "Default";

**Pull request merged**